packages/@react-spectrum/calendar/src/CalendarCell.tsx

```diff
@@ -53,6 +54,13 @@ export function CalendarCell({state, currentMonth, ...props}: CalendarCellProps)
   let {focusProps, isFocusVisible} = useFocusRing();
   let {hoverProps, isHovered} = useHover({isDisabled: isDisabled || isUnavailable || state.isReadOnly});
 
+  // Style disabled (i.e. out of min/max range), but selected dates as unavailable
+  // since it is more clear than than trying to dim the selection.
+  if (isDisabled && isInvalid) {
+    isDisabled = false;
+    isUnavailable = true;
+  }
```

Comment:
move this down into the object for classNames? then we don't edit either of the values
```
'is-disabled': isDisabled && !isInvalid,
'is-unavailable': isUnavailable || (isInvalid && isDisabled),
```